### ISSUE TITLE
High CPU and GPU usage spike with 4 tabs open

### ISSUE THREAD

**Issue by jctaillandier** [brave/brave-browser#8779]
## Description 
Happened twice, both times I had 3 tabs, open. Two youtube, one GoogleCal. At midpoint watching a video, CPU usage spiked, filled 8Gb of Ram, then started to fill the swap. Dozens of CPU and GPU process were running simultaneously and got this screenshot before shutting down.
Running Ubuntu 18.04 
Only extension installed is Full Page Screen Capture, but was not using.

## Steps to Reproduce
Not reproducible everytime
   1. Launch Browser
   2. Open a handfull of tabs, 1+ of which is youtube
   3. Run a video and monitor memory usage

## Actual result:
![brave_memory](https://user-images.githubusercontent.com/31980034/77168702-a424d900-6a8e-11ea-9ace-454d11624472.png)

## Expected result:
Not sure to me but not this ^.

## Reproduces how often: 
Intermittent

## Brave version (brave://version info)
Brave | 1.4.96 Chromium: 80.0.3987.132 (Official Build) (64-bit)
-- | --
Revision | fcea73228632975e052eb90fcf6cd1752d3b42b4-refs/branch-heads/3987@{#974}
OS | Linux
JavaScript | V8 8.0.426.26
Flash | (Disabled)
User Agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.132 Safari/537.36
Command Line | /opt/brave.com/brave/brave --enable-dom-distiller --disable-domain-reliability --no-pings --extension-content-verification=enforce_strict --extensions-install-verification=enforce --sync-url=https://no-thanks.invalid --enable-features=PasswordImport,SimplifyHttpsIndicator,WebUIDarkMode --disable-features=AutofillServerCommunication,AllowPopupsDuringPageUnload,LookalikeUrlNavigationSuggestionsUI,NotificationTriggers,AudioServiceOutOfProcess,SmsReceiver,WebXR,VideoPlaybackQuality,WebXrGamepadModule --flag-switches-begin --flag-switches-end --disable-webrtc-apm-in-audio-service --disable-sync
Executable Path | /opt/brave.com/brave/brave
Profile Path | /home/jc/.config/BraveSoftware/Brave-Browser/Default

## Version/Channel Information:

- Can you reproduce this issue with the current release?
- Can you reproduce this issue with the beta channel?
- Can you reproduce this issue with the dev channel?
- Can you reproduce this issue with the nightly channel?

**Comment by NejcZdovc** [comment]
@jctaillandier do you have rewards enabled. Is your test plan on clean profile?

**Comment by NejcZdovc** [comment]
could you also please go to Hamburger Menu --> More Tools --> Task Manager

Sort by CPU and keep an eye on CPU spikes and which tab/process is causing it.

**Comment by jctaillandier** [comment]
No Rewards enabled. 
>Is your test plan on clean profile?
I don't understand the question

Will make sure to open the task manager next time it spikes. At normal it looks like this
![task](https://user-images.githubusercontent.com/31980034/77197275-4bb80080-6abb-11ea-91fb-832090bf72d1.png)

**Comment by bsclifton** [comment]
OK folks - we've done some investigation and are fairly sure we have found the root cause and the fix  😄 Build is happening now and we'll look at releasing a hotfix. Stay tuned 😄

**Comment by bsclifton** [comment]
@jctaillandier  (or others experiencing the issue) if you wanted to try a work-around while we're prepping the hotfix, here's what you can try:
1. Visit brave://flags/#brave-adblock-cosmetic-filtering
2. Enable this option
3. You'll be prompted to restart- go ahead and do that
4. Problem should be fixed 😄

**Comment by kjozwiak** [comment]
For those users that have been experiencing various CPU/performance issues, could you please download the following version and let us know if things have improved. Once we get enough confirmations from the community, we'll push this version to our Release channel ASAP so everyone receives the update.

> * Win x64 --> https://github.com/brave/brave-browser/releases/download/v1.5.114/BraveBrowserStandaloneSetup.exe
> 
> * Win x86 --> https://github.com/brave/brave-browser/releases/download/v1.5.114/BraveBrowserStandaloneSetup32.exe
> 
> * macOS --> https://github.com/brave/brave-browser/releases/download/v1.5.114/Brave-Browser.dmg
> 
> * Linux --> https://github.com/brave/brave-browser/releases/download/v1.5.114/brave-browser_1.5.114_amd64.deb

**Comment by kjozwiak** [comment]
STR that QA can use to verify that https://github.com/brave/brave-core/pull/4999 has been fixed:

* Download & install either https://github.com/brave/brave-browser/releases/tag/v1.5.112 or https://github.com/brave/brave-browser/releases/tag/v1.5.113
* launch https://www.peteresnyder.com/brave/break.html 
* Wait ~5-10min while https://www.peteresnyder.com/brave/break.html is opened and you'll notice that Brave will start using a lot of the CPU causing performance issues.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Mint 19.1 x64` using the following build:

```
Brave | 1.5.115 Chromium: 80.0.3987.149 (Official Build) (64-bit)
--- | ---
Revision | 5f4eb224680e5d7dca88504586e9fd951840cac6-refs/branch-heads/3987_137@{#16}
OS | Linux
```

* Went through https://github.com/brave/brave-browser/issues/8779#issuecomment-602099597 and left https://www.peteresnyder.com/brave/break.html opened for ~15min without any CPU/performance issues.

![Screenshot from 2020-03-21 19-51-50](https://user-images.githubusercontent.com/2602313/77239687-0d523c80-6bb4-11ea-9eaa-5a29dee00aa1.png)

Verification **PASSED** on `macOS 10.15.3 x64` using the following build:

```
Brave | 1.5.115 Chromium: 80.0.3987.149 (Official Build) (64-bit)
-- | --
Revision | 5f4eb224680e5d7dca88504586e9fd951840cac6-refs/branch-heads/3987_137@{#16}
OS | macOS Version 10.15.3 (Build 19D76)
```

* Went through https://github.com/brave/brave-browser/issues/8779#issuecomment-602099597 and left https://www.peteresnyder.com/brave/break.html opened for ~15min without any CPU/performance issues.

<img width="1591" alt="Screen Shot 2020-03-21 at 7 34 08 PM" src="https://user-images.githubusercontent.com/2602313/77239695-1b07c200-6bb4-11ea-9576-c253c141c201.png">

Verification **PASSED** on `Win 10 x64` using the following build:

```
Brave | 1.5.115 Chromium: 80.0.3987.149 (Official Build) (64-bit)
--- | ---
Revision | 5f4eb224680e5d7dca88504586e9fd951840cac6-refs/branch-heads/3987_137@{#16}
OS | Windows 10 OS Version 1909 (Build 18363.720)
```

* Went through https://github.com/brave/brave-browser/issues/8779#issuecomment-602099597 and left https://www.peteresnyder.com/brave/break.html opened for ~15min without any CPU/performance issues.

![Annotation 2020-03-21 205812](https://user-images.githubusercontent.com/2602313/77239955-cade2f00-6bb6-11ea-92f2-56105524bacc.jpg)

**Comment by pes10k** [comment]
@kjozwiak is that file still needed (i.e. https://www.peteresnyder.com/brave/break.html)?  Can you give me a heads up when i can take it down?

**Comment by kjozwiak** [comment]
@pes10k you can take it down now 👍 We're about to release the RC right now so we won't be needing it anymore. Thanks for getting that setup. Really appreciate it.

**Comment by Huuuze** [comment]
> @pes10k you can take it down now 👍 We're about to release the RC right now so we won't be needing it anymore. Thanks for getting that setup. Really appreciate it.

I have the latest release and the page mentioned above continues to chew memory.

**Comment by fhoshino** [comment]
After updating to 1.5.115, though long term CPU hog seems to be resolved, still seeing a higher CPU usage compared to 1.4.x, especially on script heavy pages.

**Comment by atommicios** [comment]
S[]()

**Comment by jctaillandier** [comment]
Cheers lads! 
Appreciate the quick work!

**Comment by andrew-pyle** [comment]
End user here. Still having high CPU usage issues with Brave Version 1.5.115

Immediately upon opening Brave, I'm getting a Brave Browser Helper process pegged at 100% CPU.

I've also enabled "Enable cosmetic filtering" flag with no effect.

```
Brave | 1.5.115 Chromium: 80.0.3987.149 (Official Build) (64-bit)
-- | --
Revision | 5f4eb224680e5d7dca88504586e9fd951840cac6-refs/branch-heads/3987_137@{#16}
OS | macOS Version 10.14.6 (Build 18G3020)
```
![image](https://user-images.githubusercontent.com/25519964/77343353-8c07c080-6cff-11ea-8d06-fa89fe9f0f82.png)

**Comment by bsclifton** [comment]
@andrew-pyle can you please open the in-browser task manager? Basically Hamburger menu (top right) => More Tools => Task Manager

I suspect the Bat Ledger process is what is taking all the CPU. Having a screenshot would be extremely useful. You can disable the cosmetic filter - the work-around was just something which would help for 1.5.113 😄

**Comment by NejcZdovc** [comment]
@andrew-pyle do you maybe know when this started happening?

**Comment by andrew-pyle** [comment]
Disabled cosmetic filter—Thanks!

@bsclifton You are correct—Utility: Bat Ledger Service taking all the CPU.
![image](https://user-images.githubusercontent.com/25519964/77358097-3e974d80-6d17-11ea-88c8-1cf32258a4e0.png)

Is there something to be done to correct this on my end?

@NejcZdovc I've been noticing high fan usage for about 2 weeks.

**Comment by NejcZdovc** [comment]
@andrew-pyle would you be maybe able to jump on a quick call to check this one out? If so please send me an email to nejc[@]brave.com. Thank you

**Comment by jorda0mega** [comment]
![image](https://user-images.githubusercontent.com/484084/78324591-20b1c000-7543-11ea-96cc-bd92894388f6.png)
I am still getting high CPU usage that is not caused by the BAT Ledge. Any thoughts?

**Comment by fhoshino** [comment]
Same as @jorda0mega, but on Windows, not related to BAT, the browser core is taking up CPU usage, might be due to change in adblocker service.

**Comment by bsclifton** [comment]
We do have a crash fix coming - Chromium just released a minor bump (`80.0.3987.163`). It may not make an impact here. Thanks for sharing the details @jorda0mega @fhoshino

**Comment by pradeepb6** [comment]
I am also facing the CPU usage issue on Version 1.8.90 Chromium: 81.0.4044.129 (Official Build) (64-bit)
![image](https://user-images.githubusercontent.com/6407736/81164116-1d6d8200-8f90-11ea-8795-04d512ae54eb.png)

When I close the browser, laptop does not make any sound. 

My system details
![image](https://user-images.githubusercontent.com/6407736/81164296-6b828580-8f90-11ea-99e3-216e258be4af.png)
